### PR TITLE
Multiple components + quality of life

### PR DESCRIPTION
**Pull request by JohanRydin**
Server: Spring server in Java is basically untouched from the presentation. 
Dockerfiles: Added for server and for database (my researched version that I think will work)
Compose: Is working from what I can tell. You can now run docker compose up to build all the components 
Database: I have made a dockerfile for running our mysql database (as specified by lawend in the new file blueprints.sql) in docker. 

Documentation: Readmes have been somewhat uppdated. 

Next steps: 
- Add some dummies to the database so that we dont have to create them all the time (just some queries in the blueprints)
- Add the maven dependency to connect to mysql databases. This scares me. 
- Ensure that everything actually works. I only ran it once to see if it would compile. 
- Fix models for users, recommendation etc in server now that we have defined those. 
- Better Readmes.

### CODE REVIEW

File: SpringAPI/Readme.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+build: mvn clean install 
+
+run: mvn spring-boot:run 
+
+# Docker 
+
+Assuming you have Docker Engine (and Docker Desktop)
+
+docker build -t springapi:latest . # This takes 400s (TODO: Johan is naughy and downloads and entire ubuntu when there are better alternatives, will fix later)
```
Comment (Tjasomfan) [inline SpringAPI/Readme.md:16]: very naughty indeed

File: compose.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      context: ./db
+    container_name: db
+    environment:
+      MYSQL_ROOT_PASSWORD: root
+      MYSQL_DATABASE: storage
+    volumes:
+      - mysql-data:/var/lib/mysql  # Persistent storage for MySQL data
+      - ./db/blueprint.sql:/docker-entrypoint-initdb.d/blueprint.sql  # Unsure what this does but it is our entrypoint for loading our database appearance
+    ports:
+      - "3306:3306"
```
Comment (Tjasomfan) [inline compose.yaml:13]: finns det anledningar för dessa specifika portar elr tog vi bara någon randomly?

**Review by Tjasomfan — APPROVED** [review]

File: SpringAPI/Readme.md (hunk 1 above)
Comment (JohanRydin) [reply SpringAPI/Readme.md:16]: > very naughty indeed
Oops det är utdaterad info

File: compose.yaml (hunk 2 above)
Comment (JohanRydin) [reply compose.yaml:13]: 3306 är MySql:s standardport (vetefan varför, de här tomtarna ville väl ha det så: https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=mysql). För adminer tog jag en port som börjar på 4 så att autocomplete fungerar bättre :)

File: SpringAPI/Readme.md (hunk 1 above)
Comment (Finitel) [reply SpringAPI/Readme.md:16]: Too naughty

**Pull request merged**